electrum/transaction.py

```diff
@@ -1895,11 +1923,13 @@ def sign(self, keypairs) -> None:
     def sign_txin(self, txin_index, privkey_bytes, *, bip143_shared_txdigest_fields=None) -> str:
         txin = self.inputs()[txin_index]
         txin.validate_data(for_signing=True)
+        sighash = txin.sighash if txin.sighash is not None else Sighash.SIGHASH_ALL
+        sighash_type = '{0:02x}'.format(sighash)
```

Comment:
Maybe it's just personal preference, but I find this more clear:
```suggestion
        sighash_type = sighash.to_bytes(length=1, byteorder="big").hex()
```